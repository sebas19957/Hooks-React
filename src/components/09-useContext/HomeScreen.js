import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomeScreen = () => {

    const userContext = useContext(UserContext);
    const {user} = userContext;
    console.log(userContext);

    return (
        <div>
            <h1>HomeScreen</h1>
            <hr/>

            <pre>
                {JSON.stringify(user, null, 3)}
            </pre>
        </div>
    )
}
