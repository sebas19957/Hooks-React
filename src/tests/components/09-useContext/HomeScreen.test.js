import React from 'react';
import { mount, shallow } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <HomeScreen/>', () => {
   
    const user = {
        name: 'Sebastian',
        email: 'sebas1995@email.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen/>
        </UserContext.Provider>
    );

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    
});
