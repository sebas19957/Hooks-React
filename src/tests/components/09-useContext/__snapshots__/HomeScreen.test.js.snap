// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen/> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Sebastian",
   "email": "sebas1995@email.com"
}
  </pre>
</div>
`;
